src/java.desktop/share/classes/javax/swing/plaf/basic/BasicScrollBarUI.java

```diff
@@ -1201,6 +1219,28 @@ public boolean getSupportsAbsolutePositioning() {
         return supportsAbsolutePositioning;
     }
 
+    /**
+     * A listener to listen for keyboard focus changes.
+     */
+    protected class KeyboardFocusListener implements PropertyChangeListener {
+        /**
+         * Constructs a {@code KeyboardFocusListener}.
+         */
+        protected KeyboardFocusListener() {}
+
+        @Override
+        public void propertyChange(PropertyChangeEvent e) {
+            String propertyName = e.getPropertyName();
+
+            if ("focusOwner" == propertyName) {
+                // Stop scrolling if no longer focus owner
+                if (e.getNewValue() == null && scrollTimer.isRunning()) {
+                    scrollTimer.stop();
```

Comment:
In some places `scrollTimer.stop()` is followed by `scrollbar.setValueIsAdjusting(false)`. Is it applicable for KeyboardFocusListener ?
For instance:
https://github.com/openjdk/jdk/blob/0cc3caf2b49426bb6738a9792c88275c0cc7774a/src/java.desktop/share/classes/javax/swing/plaf/basic/BasicScrollBarUI.java#L1608

Reply:
You're right. I added this to my fix, but didn't get it pushed yet. Thanks for the reminder.